lib/ed25519.js

```diff
@@ -64,6 +67,67 @@ ed25519.generateKeyPair = function(options) {
   return {publicKey: pk, privateKey: sk};
 };
 
+ed25519.privateKeyFromAsn1 = function(obj) {
+  var capture = {};
+  var errors = [];
+  var valid = forge.asn1.validate(obj, privateKeyValidator, capture, errors);
+  if(!valid) {
+    var error = new Error('Invalid Key.');
+    error.errors = errors;
+    throw error;
+  }
+  var oid = forge.asn1.derToOid(capture.privateKeyOid);
+  var ed25519Oid = forge.oids.EdDSA25519;
+  if(oid !== ed25519Oid) {
+    throw new Error('Invalid OID "' + oid + '"; OID must be "' +
+      ed25519Oid + '".');
+  }
+  var privateKey = capture.privateKey;
+  // manually extract the private key bytes from nested octet string, see FIXME:
+  // https://github.com/digitalbazaar/forge/blob/master/lib/asn1.js#L542
+  var privateKeyBytes = forge.asn1.fromDer(privateKey).value;
+  // FIXME: what should be returned from this API?  Should this compose the
+  // 64 byte public key?
```

Comment:
This should be consistent with how the keys are being used elsewhere in forge. It looks like it's 64 bytes from the above constant, so we should use that. It would be very unusual for people to use forge's `fromASN1` API and get a value that they can't use with the other APIs.